File: src/main/java/noh/jinil/boot/service/UserService.java
package noh.jinil.boot.service;

import noh.jinil.boot.domain.model.UserEntity;
import noh.jinil.boot.service.shared.RegisterUserInit;
import org.springframework.security.core.userdetails.UserDetailsService;

public interface UserService extends UserDetailsService {

	UserEntity registerUser(RegisterUserInit init);

}

File: src/main/kotlin/noh/jinil/boot/service/UserService.kt
package noh.jinil.boot.service

import com.google.firebase.auth.FirebaseToken
import noh.jinil.boot.domain.entity.UserEntity
import noh.jinil.boot.service.shared.RegisterUserInit
import org.springframework.security.core.userdetails.UserDetailsService

interface UserService : UserDetailsService {
    fun registerUser(holder: FirebaseToken): UserEntity?
    fun verifyUser(holder: FirebaseToken): Boolean
}